### PR TITLE
Removed Leftover X Pack references #2517

### LINKED ISSUE #2517: [CHORE] Leftover X Pack references
**Description**

There are still some x-pack references in the code that need to be removed. 

- [ ] https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/plugins/data/README.md?plain=1#L114
- [ ] https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/packages/osd-pm/src/utils/validate_dependencies.ts#L178
- [ ] https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/core/server/saved_objects/migrations/core/index_migrator.test.ts#L145
- [ ] https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/core/server/saved_objects/migrations/core/index_migrator.test.ts#L168
- [ ] https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/core/server/saved_objects/migrations/core/opensearch_index.ts#L314
- [ ] https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/dev/notice/cli.js#L83
- [ ] https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/plugins/apm_oss/server/tutorial/instructions/apm_server_instructions.ts#L39

Credits: @AMoo-Miki 

**To Reproduce**
1. Go to the link
2. See the X Pack reference

**Expected behavior**
We should not have any X Pack references in the code

**OpenSearch Version**
Please list the version of OpenSearch being used.

**Dashboards Version**
Please list the version of OpenSearch Dashboards being used.

**Plugins**

Please list all plugins currently enabled.

**Screenshots**

**Host/Environment (please complete the following information):**
 - Browser and version [e.g. 22]

**Additional context**

### PR DESCRIPTION
**Pull request by vimalMK**
Signed-off-by: vimal K <vimalinfo10@gmail.com>

### Description
x-pack references in the code are removed as per the given files in #2517 
 
### Issues Resolved
#2517 
 
### Check List
- [ ] All tests pass
  - [x]  yarn lint
  - [x] `yarn test:jest`
  - [x] `yarn test:jest_integration`
  - [ ] `yarn test:ftr`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [x] Update [CHANGELOG.md](./../CHANGELOG.md)
- [x] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Comment by joshuarrrr** [comment]
@vimalMK Thanks for the contribution! I'll plan to take a look at this on Monday.

**Comment by vimalMK** [comment]
@joshuarrrr Thank you !

File: src/plugins/data/README.md [hunk 1]
```diff
@@ -111,7 +111,7 @@ The `SearchSource` API is a convenient way to construct and run an OpenSearch se
 
 #### Default Search Strategy
 
-One benefit of using the low-level search API, is partial response support in X-Pack, allowing for a better and more responsive user experience.
+One benefit of using the low-level search API, is allowing for a better and more responsive user experience.
```
Comment (joshuarrrr) [inline src/plugins/data/README.md:114]: I'm not sure if this is true - I may make a follow-up issue, but for now, we can make it a little more readable.

```suggestion
One benefit of using the low-level search API is that it allows for a better and more responsive user experience with partial responses.
```

**Review by joshuarrrr — COMMENTED** [review]
Thanks so much for the contribution, it looks great overall. I suggested just a couple minor cleanups of phasing.

File: src/plugins/data/README.md (hunk 1 above)
Comment (vimalMK) [reply src/plugins/data/README.md:114]: @joshuarrrr Great. Will make those changes and append to this PR. 👍

File: src/plugins/apm_oss/server/tutorial/instructions/apm_server_instructions.ts [hunk 2]
```diff
@@ -36,7 +36,7 @@ export const createEditConfig = () => ({
   }),
   textPre: i18n.translate('apmOss.tutorial.editConfig.textPre', {
     defaultMessage:
-      "If you're using an X-Pack secured version of Elastic Stack, you must specify \
+      "If you're using an secured version of Elastic Stack, you must specify \
```
[the remark this answers is not in the document]
Comment (kavilla) [reply src/plugins/apm_oss/server/tutorial/instructions/apm_server_instructions.ts:39]: +1

**Comment by kavilla** [comment]
Thank you @vimalMK would you be able to make the changes as suggested by Josh and then we can get this merged! Appreciate it so much!

**Comment by vimalMK** [comment]
> Thank you @vimalMK would you be able to make the changes as suggested by Josh and then we can get this merged! Appreciate it so much!

@kavilla Thank you. Will do the required changes.

**Comment by ashwin-pc** [comment]
@vimalMK Also dont forget to add an entry to `./changelog.md` to pass the changelog check in the CI

**Comment by vimalMK** [comment]
@kavilla @joshuarrrr PR patched with changes. 

Also @ashwin-pc for the entry to ./changelog.md, I added it under bugs because this issue is a chore and bug made more sense to me. But please let me know if needed change.

**Comment by ashwin-pc** [comment]
@vimalMK that works, thank you!

**Review by AMoo-Miki — DISMISSED** [review]

**Review by ashwin-pc — DISMISSED** [review]

**Review by joshuarrrr — APPROVED** [review]

**Comment by vimalMK** [comment]
@ashwin-pc  Resolved merge conflict in CHANGELOG.md. Apologies If I should have waited for further notice.

**Comment by ashwin-pc** [comment]
@vimalMK the changelog merge conflict isn't really your fault. A PR that we merged earlier changed the formatting causing a lot of the PR's to have a conflict. We'll address it soon. Sorry for the churn.

**Review by joshuarrrr — APPROVED** [review]

**Review by kavilla — APPROVED** [review]

**Pull request merged**